src/app/shared/services/dialog-confirmation.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Rundown } from '../../core/models/rundown'
+import { RundownMode } from '../../core/enums/rundown-mode'
+import { BasicRundown } from '../../core/models/basic-rundown'
+import { RundownService } from '../../core/abstractions/rundown.service'
+import { BasicRundownStateService } from '../../core/services/basic-rundown-state.service'
+import { DialogService } from './dialog.service'
+import { Injectable } from '@angular/core'
+import { DialogColorScheme, DialogSeverity } from '../components/confirmation-dialog/confirmation-dialog.component'
+
+@Injectable()
+export class DialogConfirmationService {
+  private readonly activationOkButtonText: string = $localize`dialog.button.activate`
+  private readonly activationMessage: string = $localize`dialog.message.activation`
+  private readonly activationWarningMessage: string = $localize`dialog.message.activation.warning`
+
+  private readonly rehearseOkButtonText: string = $localize`dialog.button.rehearse`
+  private readonly rehearseMessage: string = $localize`dialog.message.rehearse`
+  private readonly rehearseWarningMessage: string = $localize`dialog.message.rehearse.warning`
+
+  constructor(
+    private readonly dialogService: DialogService,
+    private readonly basicRundownStateService: BasicRundownStateService,
+    private readonly rundownService: RundownService
+  ) {}
+
+  public switchActivateRundownDialog(rundown: Rundown): void {
+    if (rundown.mode === RundownMode.ACTIVE) {
+      return
+    }
+
+    const nonIdleRundown: BasicRundown | undefined = this.basicRundownStateService.getNonIdleRundown()
+    if (!nonIdleRundown) {
+      this.dialogService.createConfirmDialog(rundown.name, this.activationMessage, this.activationOkButtonText, () => this.rundownService.activate(rundown.id).subscribe())
+      return
+    }
+
+    if (nonIdleRundown.mode === RundownMode.REHEARSAL && nonIdleRundown.id === rundown.id) {
+      this.dialogService.createConfirmDialog(rundown.name, this.activationMessage, this.activationOkButtonText, () => this.rundownService.activate(nonIdleRundown.id).subscribe())
+      return
+    }
+
+    this.dialogService.createConfirmDialog(
+      rundown.name,
+      `${this.activationWarningMessage} "${nonIdleRundown.name}"`,
+      `${this.activationOkButtonText}`,
+      () => this.rundownService.deactivate(nonIdleRundown.id).subscribe(() => this.rundownService.activate(rundown.id).subscribe()),
+      DialogColorScheme.DARK,
+      DialogSeverity.DANGER
+    )
+  }
+
+  public switchRehearsalRundownDialog(rundown: Rundown): void {
+    if (rundown.mode === RundownMode.REHEARSAL) {
+      return
+    }
```

Comment:
It is not expressed in the task, but from my understanding we don't won't to be able to go from an active (on air) rundown to rehearsal. Shouldn't that apply for cross-rundown mode transitions as well?
@LindvedKrvang what are your thoughts?

Reply:
That's a good question. I can see the argument for why it shouldn't be allowed. But this is a PO question, so we need to ask Jacob.

Reply:
It is actually a product team question, since we don't have a PO ;)

Reply:
So should I wait before I merge in develop, or are we good to go?

Reply:
We are good to go. We'll add it as a feature/bug task if necessary :)